features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96.cpp

```diff
@@ -57,8 +57,8 @@ static const intptr_t cellular_properties[AT_CellularBase::PROPERTY_MAX] = {
     1,  // AT_CMGF
     1,  // AT_CSDH
     1,  // PROPERTY_IPV4_STACK
-    0,  // PROPERTY_IPV6_STACK
```

Comment:
This change is causing DNS resolving problems due to the way _stack_type is currently used when BG96 socket is created.
Modem might get only IPv4 address eventhough the PDP context is IPV4V6 and because stack type is set as PDP type => _stack_type set as IPV4V6 => IP_address for the socket opening set to "0:0:0:0:0:0:0:1" => fails to send DNS requests.

_stack_type requires to be set according to the IP address(es) modem is assigned by network. IPV4 for only IPV4 address, IPV6 for only IPV6 address, IPV4V6 when both IPV4 and IPV6 address are assigned by network.

Some kind of fix is needed before we cand do these stack PROPERTY changes since they are breaking the DNS resolving?

Reply:
If nothing else changes, at least (_stack_type == IPV6_STACK) ? "0:0:0:0:0:0:0:1":"127.0.0.1" might be better check than
(_stack_type == IPV4_STACK) ? "127.0.0.1" : "0:0:0:0:0:0:0:1".

Reply:
Anything to address here? I set this to needs: review just in case

Reply:
Once answered (all good?), we can merge

Reply:
@mirelachirica Please re-review, no more property flag changes in this PR.